tests/test_to_container.py

```diff
@@ -167,3 +167,121 @@ def test_string_interpolation_with_readonly_parent() -> None:
 def test_to_container_missing_inter_no_resolve(src: Any, expected: Any) -> None:
     res = OmegaConf.to_container(src, resolve=False)
     assert res == expected
+
+
+class TestEnumToStr:
+    """Test the `enum_to_str` argument to the `OmegaConf.to_container function`"""
+
+    @pytest.mark.parametrize(
+        "src,enum_to_str,expected",
+        [
+            pytest.param(
+                DictConfig(content={Color.RED: "enum key"}),
+                True,
+                "RED",
+                id="enum_key:convert",
+            ),
+            pytest.param(
+                DictConfig(content={Color.RED: "enum key"}),
+                False,
+                Color.RED,
+                id="enum_key:dont-convert",
+            ),
+            pytest.param(
+                DictConfig(content={123: "int key"}),
+                True,
+                123,
+                id="int_key:T-unaffected",
+            ),
+            pytest.param(
+                DictConfig(content={123: "int key"}),
+                False,
+                123,
+                id="int_key:F-unaffected",
+            ),
+        ],
+    )
+    def test_enum_to_str_for_keys(
+        self, src: DictConfig, enum_to_str: bool, expected: Any
+    ) -> None:
+        container = OmegaConf.to_container(src, enum_to_str=enum_to_str)
+        assert isinstance(container, dict)
+        key = list(container.keys())[0]
```

Comment:
If you use the same key name in all inputs you would be able to use it directly here.

Reply:
Done in https://github.com/omry/omegaconf/pull/549/commits/910e4d9df9acfee9738ab3226199b855936a13d0.